### ISSUE TITLE
LLVM Version Harmonization

### ISSUE THREAD

**Issue by Victor-Jung** [pulp-platform/Deeploy#19]
Currently, we use LLVM 12 to compile applications for the Snitch platform, while we use LLVM 15 for the rest. We should investigate if there is a strong reason not to use LLVM 15 everywhere.

**Comment by FrancescoConti** [comment]
Does it make sense to do this and then switch afterwards to LLVM 18 when it's ready? Or better to do everything in one step?

**Comment by Victor-Jung** [comment]
Good point; I think it's best to try harmonizing the version when we move to 18.
